lib/puma/io_buffer.rb

```diff
@@ -1,4 +1,19 @@
 # frozen_string_literal: true
 
-require 'puma/detect'
-require 'puma/puma_http11'
+module Puma
+  class IOBuffer < String
+    def initialize
+      if RUBY_VERSION <= "2.3"
```

Comment:
This condition is incorrect (e.g. `2.3.4` <= `2.3` doesn't hold), `set capacity if RUBY_VERSION >= 2.4` should work fine instead, and that's likely what makes the 2.3 build fails:
https://github.com/puma/puma/pull/1980/checks?check_run_id=258129791
```
/home/runner/work/puma/puma/lib/puma/io_buffer.rb:9:in `initialize': unknown keyword: capacity (ArgumentError)
	from /home/runner/work/puma/puma/lib/puma/io_buffer.rb:9:in `initialize'
	from /home/runner/work/puma/puma/lib/puma/thread_pool.rb:97:in `new'
	from /home/runner/work/puma/puma/lib/puma/thread_pool.rb:97:in `block (2 levels) in spawn_thread'
	from /home/runner/work/puma/puma/lib/puma/thread_pool.rb:97:in `map'
	from /home/runner/work/puma/puma/lib/puma/thread_pool.rb:97:in `block in spawn_thread'
```